backend/routers/referral.py

```diff
@@ -0,0 +1,44 @@
+from fastapi import APIRouter, Depends, Header, HTTPException
+from models.database import SessionLocal
+from sqlalchemy.orm import Session, joinedload
+import models.referral as model
+import schemas.referral as schema
+
+router = APIRouter()
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+@router.get("/referrals")
+async def get_prescription(patient_id = Header(1), db: Session = Depends(get_db)):
+    response = db.query(model.Referral).filter(model.Referral.patient_id == patient_id).all()
+    return response
+    
+@router.get("/referals/{referal_id}")
+async def get_prescription(referal_id: int, db: Session = Depends(get_db)):
+    referal = db.query(model.Referral).filter(model.Referral.id == referal_id).first()
+    if referal is None:
+        raise HTTPException(status_code=404, detail="Referal not found")
+    return referal
+
+@router.post("/referal")
+async def add_prescription(referal: schema.Referral, patient_id = Header(1), doctor_id = Header(1), db: Session = Depends(get_db)):
```

Comment:
Data should be read from request body not header

Reply:
bump

Reply:
You're right